nova_plugin/server.py

```diff
@@ -548,11 +548,21 @@ def attach_volume(nova_client, cinder_client, **kwargs):
         server_id,
         volume_id,
         device if device != 'auto' else None)
-    volume.wait_until_status(cinder_client=cinder_client,
-                             volume_id=volume_id,
-                             status=volume.VOLUME_STATUS_IN_USE)
-    if device == 'auto':
+    try:
+        vol, wait_succeeded = volume.wait_until_status(
+            cinder_client=cinder_client,
+            volume_id=volume_id,
+            status=volume.VOLUME_STATUS_IN_USE
+        )
+        if not wait_succeeded:
+            raise RecoverableError(
+                'Waiting for volume status {0} failed - detaching volume and '
+                'retrying..'.format(volume.VOLUME_STATUS_IN_USE))
+    except RecoverableError:
```

Comment:
2) I have already fixed it.

As for the 1st comment, I had taken it under consideration, but the problem is that my "cleanup" begins with the same API call, which will likely fail immediately after the first call fails..